zingo-cli/src/lib.rs

```diff
@@ -516,37 +516,44 @@ fn dispatch_command_or_start_interactive(cli_config: &ConfigTemplate) {
     if cli_config.command.is_none() {
         start_interactive(command_transmitter, resp_receiver);
     } else {
-        // Optionally wait for background sync to finish before executing command
+        // Optionally wait for background sync to finish before executing command.
+        // Retry requesting status until the channel closes or the percentage field
+        // indicates completion.
         if cli_config.sync && cli_config.waitsync {
             use std::{thread, time::Duration};
+
             loop {
-                // Poll sync task status
+                // Request machine-readable sync status.
                 command_transmitter
-                    .send(("sync".to_string(), vec!["poll".to_string()]))
```

Comment:
sync poll cannot be replaced by sync status. sync poll is polling the sync handle and it the only way to return the actual errors from the sync task.